roles/users/tasks/main.yml

```diff
@@ -50,7 +50,7 @@
   changed_when: false
   sudo: no
   register: admin_user_status
-  when: not sshd_permit_root_login
+  when: not sshd_permit_root_login and ansible_ssh_host is defined
```

Comment:
If staging or production hosts files are like the Trellis default files and list only IP and do not explicitly define `ansible_ssh_host`, I'm pretty sure `ansible_ssh_host is defined` will return `false`. If so, this task will sometimes not run for staging/production but should. Does that sound right?

Anyone aware of an ansible variable to check whether `--connection=local` was passed? That strikes me as a more direct test of the issue at hand. @chriszarate What specifically fails when Packer runs this? Is it that the ping module run on a local connection always fails? Just fails for admin user? Can we test for that specific problem instead of just testing local connection (but perhaps local connection is the exact problem)?